wrfhydro/example/wrfhydro_simulation.py

```diff
@@ -0,0 +1,650 @@
+        return f"/mnt/{drive_letter}{unix_part}"
+
+    def generate_simulation_commands(
+            self,
+            command_notes: str,
+            log_name: str
+    ) -> None:
+        """
+        Definition:
+            A function to design common commands to generate domain files
+        References:
+            https://stackoverflow.com/questions/57693460/using-wsl-bash-from-within-python
+            https://stackoverflow.com/questions/78219632/run-a-linux-application-on-wsl-from-a-windows-python-script
+            https://forum.freecad.org/viewtopic.php?t=91659
+        Arguments:
+            command_notes (str):
+                A command to be executed in WSL
+            log_name (str):
+                A name of the log file
+        """
+        # Full command notes that change the directory into the wsl_path
+        # and are executed to obtain the results
+        command_execution = f"cd {self.convert_window_to_wsl_path()} && {command_notes}"
+
+        # Execute the full command in Window Subsystem for Linux (WSL)
+        # Explanation:
+        # - ["wsl", "bash", "-c", command]: The command list to run in the shell.
+        #   * "wsl": Calls the WSL subsystem (to run Linux commands from Windows)
+        #   * "bash": Runs the Bash shell inside WSL.
+        #   * "-c": Tells Bash to execute the string following it as a command.
+        #   * command: A string containing the actual Linux command you want to run
+        #              (e.g., "ls -la" or "./geogrid.exe")
+        # - stdout=subprocess.PIPE: To capture the standard output (anything the command prints normally)
+        # - stderr=subprocess.PIPE: To capture the standard error (error messages from the command)
+        # - text=True: Tells python to return output as strings (not bytes)
+        result = subprocess.run(
+            ["wsl", "bash", "-c", command_execution],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            check=True
+        )
+
+        # Print out log file after generating geo_em netcdf file
+        with open(fr"{self.domain_path}\{log_name}.log", "w", encoding="utf-8") as f:
+            f.write("*** RETURNCODE ***\n")
+            f.write(f"Return code: {result.returncode}\n")
+            f.write("*** STDOUT ***\n")
+            f.write(result.stdout)
+            f.write("*** STDERR ***\n")
+            f.write(result.stderr)
+
+    def generate_simulation_files(self) -> None:
+        """
+        Definition:
+            A function to run WRF-Hydro simulation
+        References:
+            https://wrf-hydro.readthedocs.io/en/latest/appendices.html
+            https://ral.ucar.edu/sites/default/files/public/HowToBuildandRunWRF-HydroV5inStandaloneMode_0.pdf
+        Arguments:
+            Already defined above.
+        """
+        # Define command notes and log name
+        command_notes = r"mpirun -np 2 ./wrf_hydro_NoahMP.exe >> run.log 2>&1"
```

Comment:
Have you experimented with higher values of `-np`? I think it is the multithreading number of processes variable